### ISSUE TITLE
Yari normalizing heading IDs to all lower case is breaking links

### ISSUE THREAD

**Issue by chrisdavidmills** [mdn/yari#2524]
I was just looking at this issue: https://github.com/mdn/content/issues/1297

TL;DR is that in https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events

We've got a heading in the source with a capital letter at the beginning of its ID: https://github.com/mdn/content/blob/main/files/en-us/web/guide/events/creating_and_triggering_events/index.html#L44

And document fragment link to it here: https://github.com/mdn/content/blob/main/files/en-us/web/guide/events/creating_and_triggering_events/index.html#L28

Both have identical text, so the link should work.

However, on the final page rendering, the heading's ID has been normalized to all lower case:

```
<a href="#the_old-fashioned_way" title="Permalink to The old-fashioned way">The old-fashioned way</a>
```

So the link no longer works.

A lot of links are probably broken because of this. Do we have to normalize heading IDs to all lower case?

**Comment by myfreeer** [comment]
How about listening to the [hashchange](https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event) event and [replace](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace) the hash to lower case again?

**Comment by mattiapontonio** [comment]
> Do we have to normalize heading IDs to all lower case?

I think yes manually or not.
In content.
Or revert the change here.
Don't [Overengineering](https://en.m.wikipedia.org/wiki/Overengineering#:~:text=Overengineering%20(or%20over-engineering%2C,be%20unnecessarily%20complex%20or%20inefficient.).
That's what it is, they have to match.

**Comment by peterbe** [comment]
The solution is to consider it as a flaw. Similar to how we treat broken links. 
When the flaw detection scrutinies `<a>` tags it doesn't yet do anything if the link has a `#Anchor`. 
That way all links will work and we will soon be squared away. 
For now, until we have the updated flaw detection code, we can do some search-and-replace.

**Comment by chrisdavidmills** [comment]
Dealing with this as a flaw sounds reasonable, thanks for the thoughts @peterbe !

**Comment by peterbe** [comment]
The task here is to write a flaw check (within the `broken_link` thing) that looks something like this:
```javascript
if ($a.attr('href').startsWith('#')) {
  if ($a.attr('href') !== $a.attr('href').toLowerCase()) {
    suggestion = $a.attr('href').toLowerCase()
    addFlaw($a, suggestion)
  }
} else if  ...existing code...
```
